apps/openmw/mwsound/soundmanagerimp.cpp

```diff
@@ -368,22 +368,34 @@ namespace MWSound
 
             mMusicFiles[mCurrentPlaylist] = filelist;
 
+            // Build mMusicToPlay for this playlist
+            tracklist.resize(filelist.size());
+            std::iota(tracklist.begin(), tracklist.end(), 0);
         }
         else
             filelist = mMusicFiles[mCurrentPlaylist];
 
         if(filelist.empty())
             return;
 
-        int i = Misc::Rng::rollDice(filelist.size());
+        // Do a Fisher-Yates shuffle
+        int i = Misc::Rng::rollDice(tracklist.size());
```

Comment:
Seems to me the roll should be done after the 'Repopulate if playlist is empty', or else the first track in the playlist will be more likely to play since it will always play just before a re-shuffle.